### PR TITLE
csvq: init at 1.17.11

### LINKED ISSUE #212595: Packaging request for csvq
**csvq**
From the homepage:
> Csvq is a command line tool to operate CSV files. You can read, update, delete CSV records with SQL-like query.
> 
> You can also execute multiple operations sequentially in managed transactions by passing a procedure or using the interactive shell. In the multiple operations, you can use variables, cursors, temporary tables, and other features.

**Metadata**

* homepage URL: https://mithrandie.github.io/csvq/
* source URL: https://github.com/mithrandie/csvq
* license: mit
* platforms: unix, linux, darwin, bsd, netbsd, openbsd, (windows)

**Note**

I'm just getting started with nix and don't have a lot of experience with the nix DSL. If I find time I could look into this myself but some pointers where to start when submitting new packages are appreciated. If someone else is interested in doing this, that's also fine for me (and will probably be a lot quicker :smile:)

### PR DESCRIPTION
**Pull request by tomodachi94**
###### Description of changes

Add [`csvq`](https://mithrandie.github.io/csvq/), a query language for CSV.

Closes #212595.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/csvq/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ buildGoModule, fetchFromGitHub, lib }:
+
+buildGoModule rec {
+  pname = "csvq";
+  version = "1.17.1";
```
Comment (pinn3) [inline pkgs/development/tools/csvq/default.nix:5]: Why not use the latest version? ([`1.17.11`](https://github.com/mithrandie/csvq/releases/tag/v1.17.11))

Comment (tomodachi94) [reply pkgs/development/tools/csvq/default.nix:5]: > Why not use the latest version? ([`1.17.11`](https://github.com/mithrandie/csvq/releases/tag/v1.17.11))

Typo, fixing.

Comment (tomodachi94) [reply pkgs/development/tools/csvq/default.nix]: @pinn3 fixed in the latest push.

File: pkgs/development/tools/csvq/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+buildGoModule rec {
+  pname = "csvq";
+  version = "1.17.11";
+
+  src = fetchFromGitHub {
+    owner = "mithrandie";
+    repo = "csvq";
+    rev = "v${version}";
+    sha256 = "sha256-cH4uUi/QhOxf9aXh3wRozSKPTd13PnxydKFoG32MIxM=";
```
Comment (LeSuisse) [inline pkgs/development/tools/csvq/default.nix:11]: I get a mismatch, I'm guessing this was not updated after the 1.17.1 -> 1.17.11 update.

File: pkgs/development/tools/csvq/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  };
+
+  vendorSha256 = "sha256-BIa5vAEIDNdD+xW9z7geyBJnSbOeXxCl0aHIuAMBIgE=";
+
+  doCheck = false;
```
Comment (LeSuisse) [inline pkgs/development/tools/csvq/default.nix:16]: Do we need to disable the tests?

File: pkgs/development/tools/csvq/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+  meta = with lib; {
+    description = "SQL-like query language for CSV";
+    homepage = "https://mithrandie.github.io/csvq/";
+    license = licenses.mit;
+    maintainers = with maintainers; [ tomodachi94 ];
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (LeSuisse) [inline pkgs/development/tools/csvq/default.nix:23]: I think we can drop this `platforms` attribute to use the default one of `buildGoModule`, it does not like there is something specific to the platform.

**Review by LeSuisse — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/csvq/default.nix (hunk 4 above)
Comment (tomodachi94) [reply pkgs/development/tools/csvq/default.nix]: Done.

File: pkgs/development/tools/csvq/default.nix (hunk 3 above)
Comment (tomodachi94) [reply pkgs/development/tools/csvq/default.nix]: I copied this from another Go package and forgot to remove this line.

File: pkgs/development/tools/csvq/default.nix (hunk 2 above)
Comment (tomodachi94) [reply pkgs/development/tools/csvq/default.nix]: Bumped both the regular and vendored SHAs.

**Review by LeSuisse — APPROVED** [review]

**Pull request merged**